csharp/lib/src/lib.rs

```diff
@@ -118,7 +119,11 @@ pub extern "C" fn set(
             let client = Box::leak(Box::from_raw(ptr_address as *mut Client));
             match result {
                 Ok(_) => (client.success_callback)(callback_index, std::ptr::null()), // TODO - should return "OK" string.
-                Err(_) => (client.failure_callback)(callback_index), // TODO - report errors
+                Err(err) => {
+                    logger_core::log_debug("command error", format!("callback {}, error {}, kind {:?}, code {:?}, category {:?}, detail {:?}", callback_index, err, err.kind(), err.code(), err.category(), err.detail()));
+                    let c_err_str = CString::new(error_message(&err)).expect("CString::new failed");
```

Comment:
nit: I don't think we need expect here. We could probably use unwrap instead and it would be just as helpful wrt debug info. Applies throughout.

Reply:
Fixed in 93186f3e